LinqFaster/Aggregate.cs

```diff
@@ -9,149 +13,76 @@ namespace JM.LinqFaster
     /// </summary>
     public static partial class LinqFaster
     {
-
-        // ------------------------------ Arrays --------------------------
+        // Note: Lists can have items added and removed whilst these API's are in use
+        // The IReadOnlyList<T> represents a list in which the _number_ and _order_ of list elements is read-only.
+        //
 
         /// <summary>
-        /// Applies an accumulator function over an array.
+        /// Applies an accumulator function over an array[T] / List{T} / or IReadOnlyList{T}.
         /// </summary>        
-        /// <param name="source">An array to aggregate over.</param>
+        /// <param name="source">An IReadOnlyList to aggregate over.</param>
         /// <param name="func">An accumulator function to be invoked on each element</param>
         /// <returns>The final accumulator value</returns>
-        public static TSource AggregateF<TSource>(this TSource[] source, Func<TSource, TSource, TSource> func)
```

Comment:
Converting T[] to IReadOnlyList<T> makes it have to call through the interfaces for everything, which involves a vtable lookup. If this is a change (and not a reformatting artifact of moving things etc), then this would almost surely regress performance for arrays.